src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -3720,6 +3720,18 @@ internal uint GetValEscape(BoundExpression expr, uint scopeOfTheContainingExpres
 
                     return escape;
 
+                case BoundKind.DynamicObjectCreationExpression:
```

Comment:
>case BoundKind.DynamicObjectCreationExpression [](http://example.com/codeflow?start=16&length=46)

Do we need to handle other forms of dynamic operations?

Reply:
I don't think so. I tried to but wasn't able to reproduce the error with dynamic operations other than object creation.